worker/methods.go

```diff
@@ -238,6 +238,52 @@ func (w *Worker) TaskList() (tasks []TaskInfo, err error) {
 	return out["tasks"], nil
 }
 
+type TaskListParams struct {
+	CodeName string
+	Page     int
+	PerPage  int
+	FromTime time.Time
+	ToTime   time.Time
+	Statuses []string
+}
+
+func (w *Worker) FilteredTaskList(params TaskListParams) (tasks []TaskInfo, err error) {
+	out := map[string][]TaskInfo{}
+	url := w.tasks()
+
+	if params.CodeName != "" {
```

Comment:
> I just don't want an empty parameter to be included in the request.

what's wrong with empty parameters in request? It is regular case for HTTP. Let worker's API handle all the errors ( ;

> it's a required parameter

exactly. But in the client library it looks like optional.